engine/src/main/java/org/hibernate/validator/internal/engine/resolver/TraverseAllTraversableResolver.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.engine.resolver;
+
+import java.lang.annotation.ElementType;
+
+import javax.validation.Path;
+import javax.validation.Path.Node;
+import javax.validation.TraversableResolver;
+
+
+/**
+ * {@code TraversableResolver} considering that all properties are reachable and cascadable.
+ * <p>
+ * This is the default behavior if JPA is not detected in the classpath.
+ *
+ * @author Guillaume Smet
+ */
+public class TraverseAllTraversableResolver implements TraversableResolver {
+
+	public static final TraverseAllTraversableResolver INSTANCE = new TraverseAllTraversableResolver();
```

Comment:
Having a static instance might be a bit counter productive if we use a custom `TraversableResolver` as we instantiate an object that won't be used. But I did it so we can use an equality check instead of an `instanceof` when wrapping the `TraversableResolver` with a caching one.